test/services/duplicate_rule_resolver_test.rb

```diff
@@ -49,10 +49,9 @@ class DuplicateRuleResolverTest < ActiveSupport::TestCase
 
   test 'resolves rule_results from a duplicate rule' do
     assert_difference('RuleResult.count' => 2) do
-      (rr = rule_results(:one).dup).save(validate: false)
-      (rr2 = rule_results(:one).dup).save(validate: false)
-      rules(:one).rule_results << rr
-      @dup_rule.rule_results << rr2
+      rule_results(:one).dup.update!(rule: rules(:one), host: hosts(:two),
+                                     test_result: test_results(:two))
+      rule_results(:one).dup.update!(rule: @dup_rule, host: hosts(:two))
```

Comment:
This test wasn't working anymore due to the new DB constraints. Updating the associated test_result and host to be different from `rule_results(:two)` fixture was enough to fix this test.